Tribler/Core/Modules/restapi/downloads_endpoint.py

```diff
@@ -90,3 +107,87 @@ def render_GET(self, request):
                              "max_download_speed": download.get_max_speed(DOWNLOAD)}
             downloads_json.append(download_json)
         return json.dumps({"downloads": downloads_json})
+
+
+class DownloadSpecificEndpoint(DownloadBaseEndpoint):
+    """
+    This class is responsible for dispatching requests to perform operations in a specific discovered channel.
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+
+        child_handler_dict = {"remove": DownloadRemoveEndpoint, "stop": DownloadStopEndpoint,
+                              "resume": DownloadResumeEndpoint}
+        for path, child_cls in child_handler_dict.iteritems():
+            self.putChild(path, child_cls(session, bytes(infohash.decode('hex'))))
+
+
+class DownloadRemoveEndpoint(DownloadBaseEndpoint):
+    """
+    A DELETE request to this endpoint removes a specific download from Tribler. You can specify whether you only
+    want to remove the download or the download and the downloaded data using the remove_data parameter.
+
+    Example request:
+    {
+        "remove_data": True
+    }
+    """
+
+    def __init__(self, session, infohash):
+        DownloadBaseEndpoint.__init__(self, session)
+        self.infohash = infohash
+
+    def render_DELETE(self, request):
+        parameters = http.parse_qs(request.content.read(), 1)
+
+        if 'remove_data' not in parameters or len(parameters['remove_data']) == 0:
```

Comment:
I've added the `len` check since `parameters['remove_data']` turned out to be an array and I was not sure whether this list is always filled with data. If it's not, an exception would be thrown when trying to access an empty list. So I added it to make 100% sure that we actually have the data.

I just tried make the list empty using some curl statements but it's always filled, even in the following command: `curl -X PUT --data "name=&description=" http://localhost:8085/mychannel`. @whirm do you think the check should be removed?